deploy-service/common/src/main/java/com/pinterest/arcee/aws/EC2HostInfoDAOImpl.java

```diff
@@ -99,12 +106,14 @@ public void terminateHost(String hostId) throws Exception {
         RunInstancesRequest request = new RunInstancesRequest();
         request.setImageId(groupBean.getImage_id());
         request.setInstanceType(groupBean.getInstance_type());
-        request.setKeyName("ops");
+        request.setKeyName(vmKeyName);
         request.setSecurityGroupIds(Arrays.asList(groupBean.getSecurity_group()));
         request.setSubnetId(subnet);
-        request.setUserData(groupBean.getUser_data());
+        String userData = Base64.encodeBase64String(groupBean.getUser_data().getBytes());
```

Comment:
Just to double check: the userdata we read from groupbean is always decoded string, right?